WalletWasabi/Wallets/Wallet.cs

```diff
@@ -257,6 +262,36 @@ public override async Task StartAsync(CancellationToken cancel)
 		}
 	}
 
+	private async Task PerformFinalSynchronizationAsync(CancellationToken cancel)
+	{
+		using (await HandleFiltersLock.LockAsync(cancel).ConfigureAwait(false))
+		{
+			while (!cancel.IsCancellationRequested)
+			{
+				try
+				{
+					await PerformWalletSynchronizationAsync(SyncType.NonTurbo, cancel).ConfigureAwait(false);
+					if (LastProcessedFilter is { } lastProcessedFilter)
+					{
+						SetFinalBestHeight(new Height(lastProcessedFilter.Header.Height));
+					}
+
+					break;
+				}
+				catch (Exception ex)
```

Comment:
This seems dangerous to me. First, do not catch all exceptions but only those that can be handled. Ignoring problems only creates bigger problems.